src/common/reactNativeProjectHelper.ts

```diff
@@ -3,38 +3,52 @@
 
 import * as Q from "q";
 import * as semver from "semver";
-import {Package} from "./node/package";
+import {CommandExecutor} from "./commandExecutor";
 
 export class ReactNativeProjectHelper {
     private workspaceRoot: string;
-    private static REACT_NATIVE_NPM_LIB_NAME = "react-native";
 
     constructor(workspaceRoot: string) {
         this.workspaceRoot = workspaceRoot;
     }
 
+    public getReactNativeVersion(): Q.Promise<any> {
+        let deferred = Q.defer<any>();
+
+        let outputStream = new CommandExecutor(this.workspaceRoot).spawnReactCommand("-v").stdout;
+        let output = "";
+
+        outputStream.on("data", (chunk: Buffer) => {
+          output += chunk.toString();
+        });
+
+        outputStream.on("end", () => {
+            deferred.resolve(output.match(/react-native: ([\d\.]+)/));
```

Comment:
I would suggest resolving with the matching group, if present:

```
const match = output.match(/react-native: ([\d\.]+)/);
deferred.resolve(match && match[1]);
```

That way you can set the return type to `Q.Promise<string>` and avoid the error you have in `validateReactNativeVersion` where right now (I believe) you are comparing the "match" object with the version string.